core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -2132,6 +2151,9 @@ protected String getNameInSpaceForUserName(String userName) throws UserStoreExce
     protected String getNameInSpaceForUserName(String userName, String searchBase, String searchFilter) throws UserStoreException {
         boolean debug = log.isDebugEnabled();
 
+        if (userName == null) {
+            throw new UserStoreException("userName value is null.");
+        }
```

Comment:
This username null check is done mainly to handle the NPE in userCache ConcurrentHashMaps.